writeToStream.js

```diff
@@ -272,7 +272,7 @@ function publish (opts, stream) {
   }
 
   // Get the payload length
-  if (!Buffer.isBuffer(payload)) length += Buffer.byteLength(payload)
+  if (Buffer.isBuffer(payload)) length += Buffer.byteLength(payload)
```

Comment:
we need to call this for `Buffer.isBuffer(str)`? `str`  could be `'utf-8'`.

Reply:
Perhaps checking if type is string first?
(My use case always uses binaries.)

Reply:
What do you mean by binary?

Reply:
It's protobuf, so a buffer/array of "random" values that cannot be converted into strings

Reply:
> It's protobuf, so a buffer/array of "random" values that cannot be converted into strings

What type is that?

Reply:
It's a Uint8Array